### PR TITLE
Add support for different date validation formats

### PR DESCRIPTION
**Pull request by eyalvo-matrix**
I changed the validation logic to support all date formats 

one can use the validation by setting 
options.format
OR
var g_date_format='<?php echo $GLOBALS["date_display_format"];?>';

### CODE REVIEW

**Comment by eyalvo-matrix** [comment]
This also fixes a problem that format DD/MM/YYYY is not supported in the creation of a new appointment in the calendar

File: interface/main/calendar/add_edit_event.php [hunk 1]
```diff
@@ -2028,6 +2028,8 @@ function are_days_checked(){
 * this enable to add new rules for this form in the pageValidation list.
 * */
 var collectvalidation = <?php echo($collectthis); ?>;
+var g_date_format='<?php echo $GLOBALS["date_display_format"];?>';
```
Comment (bradymiller) [inline interface/main/calendar/add_edit_event.php]: hi @eyalvo-matrix ,
Isn't this already being set in validation_script.js.php that is included in this script?

Comment (eyalvo-matrix) [reply interface/main/calendar/add_edit_event.php]: As I can see add_edit_event.php only loads  validate_extend.js and validate_modified.js

Comment (bradymiller) [reply interface/main/calendar/add_edit_event.php]: I'm seeing it at line 106

Comment (eyalvo-matrix) [reply interface/main/calendar/add_edit_event.php]: Yes I see that, and It is hard to track the code but as far that I can see it is not defined in the popup
I ran the following test 
image

Comment (bradymiller) [reply interface/main/calendar/add_edit_event.php]: Looks like there is a bug in validation_script.js.php that needs to be addressed. If I look at the source from browser I see:
``````
var g_date_format='0';
``````

Comment (bradymiller) [reply interface/main/calendar/add_edit_event.php]: oh wait, actually, that is correct, it should be set to 0

Comment (bradymiller) [reply interface/main/calendar/add_edit_event.php]: So, the value is getting set in the current codebase.

Comment (eyalvo-matrix) [reply interface/main/calendar/add_edit_event.php]: Ummm ...Strange I will do a new installation and re-check it 
maybe somehow I did not get the latest pull

Comment (bradymiller) [reply interface/main/calendar/add_edit_event.php]: I think it is an ordering issue in validation_script.js.php
Note that validate_extend.js is being included before the g_date_format variable is set

**Comment by eyalvo-matrix** [comment]
Ok , I did a fresh install and indeed I see that it is defined. I think the problem is that  we do not have the latest pull or I have some kind of cache . sorry for the mess

**Comment by bradymiller** [comment]
No prob and thanks for this fix! I'll bring it into the codebase now.
-brady

**Pull request merged**